src/codeu/chat/client/View.java

```diff
@@ -280,4 +283,25 @@ public Uuid getUserGeneration() {
 
     return messages;
   }
+
+  public PublicKey getServerPublicKey() {
+    try (final Connection connection = source.connect()) {
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.GET_SERVER_PUBLIC_KEY);
+
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.GET_SERVER_PUBLIC_KEY) {
+        KeyFactory keyFactory = KeyFactory.getInstance(Serializers.STRING.read(connection.in()));
+        byte[] keyBytes = Serializers.BYTES.read(connection.in());
+        return keyFactory.generatePublic(new X509EncodedKeySpec(keyBytes));
+      } else {
+        LOG.error("Response from client setup failed.");
+      }
+
+    } catch (Exception ex) {
+      System.out.println("ERROR: Exception during client setup. Check log for details.");
+      LOG.error(ex, "Exception during client setup.");
```

Comment:
Is the error recoverable?

If not, I would recommend to exit the program on errors.